pkg/fileutils/fileutils.go

```diff
@@ -296,15 +296,15 @@ func EnsureParentDirectoryExist(fileName string) error {
 
 // EnsureDirectoryExists check if the passed directory exists or not, and if
 // it doesn't exist, create it using 0700 as permissions bits
-func EnsureDirectoryExists(destinationDir string) error {
-	if _, err := os.Stat(destinationDir); os.IsNotExist(err) {
+func EnsureDirectoryExists(destinationDir string) (err error) {
```

Comment:
while we're modifying this function, I think we should ensure that if the directory exists, it should have the right permissions. Going to try adding tests and modify.